zio-http/src/main/scala/zhttp/http/HttpData.scala

```diff
@@ -54,7 +54,7 @@ sealed trait HttpData[-R, +E] { self =>
    */
   private[zhttp] def unsafeSize: Long = self match {
     case HttpData.Empty               => 0L
-    case HttpData.Text(text, _)       => text.length.toLong
+    case HttpData.Text(text, charSet) => text.getBytes(charSet).size.toLong
     case HttpData.BinaryChunk(data)   => data.size.toLong
     case HttpData.BinaryByteBuf(data) => data.readableBytes().toLong
```

Comment:
```suggestion
    case HttpData.Complete(data) => data.readableBytes().toLong
```

Reply:
Done. I assumed we are renaming BinaryByteBuf to Complete everywhere.